crates/oxc_codegen/src/sourcemap_builder.rs

```diff
@@ -22,21 +23,21 @@ pub struct LineOffsetTable {
 }
 
 #[allow(clippy::struct_field_names)]
-pub struct SourcemapBuilder {
+pub struct SourcemapBuilder<'a> {
     enable_sourcemap: bool,
     source_id: u32,
     original_source: Arc<str>,
     last_generated_update: usize,
     last_position: Option<u32>,
     last_search_line: usize,
     line_offset_tables: Vec<LineOffsetTable>,
-    sourcemap_builder: oxc_sourcemap::SourceMapBuilder,
+    sourcemap_builder: oxc_sourcemap::SourceMapBuilder<'a>,
     generated_line: u32,
     generated_column: u32,
 }
 
-impl Default for SourcemapBuilder {
-    fn default() -> Self {
+impl<'a> SourcemapBuilder<'a> {
+    pub fn new (allocator: &'a Allocator) -> Self {
```

Comment:
You don't need to pass in the allocator, the data is self contained.

I think you can fix this by wrapping the data structures.